src/main/java/com/kodilla/ecommercee/controller/SessionController.java

```diff
@@ -0,0 +1,28 @@
+package com.kodilla.ecommercee.controller;
+
+import com.kodilla.ecommercee.domain.dto.SessionDto;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.Random;
+
+@RestController
+@RequestMapping("/Session")
+@CrossOrigin("*")
+public class SessionController {
+
+    @PostMapping(value = "{userId}")
+    public ResponseEntity<SessionDto> createSession(@PathVariable long userId) {
+        SessionDto sessionDto = new SessionDto(true,2345L);
+        return ResponseEntity.ok(sessionDto);
+    }
+
+    @PutMapping(value = "genUserKey/{userId}")
```

Comment:
Is this endpoint necessary? Maybe generating user key should be included in endpoint creatingSession so when the session is created the key is automatically also created or am I thinking wrong?

Reply:
On save site you might have both options. You might generate userKey when you create session and you might need it independently. It will happen in DbService method anyway I think. I'm not sure how this relation Session-User should work. When you are logged in user it is simple, but how it works when you are, for example guest user without an account? Or when you browsing site? Do you need an open session then or not?

Reply:
Our ERD diagram which we are working on has only option of logging in then adding items to cart and making order. Guests(not logged users) cannot add items to cart from what I understood during our yesterday metting.
Ok maybe this endpoint is valiable because I can imagine a store when we log in we got 60minutes to buy items when we want to extend the time because it almost expires we click refresh time and this leads to this endpoint and the session got new userKey which i valid for new 60minutes.
So if we think like this it is needed then.

Reply:
Maybe. And it could take UserAddDto with username and password, validate password and then generate key and return this key ... or return SessionDto? What is the difference between userKey and sessionId? I don't know.

Reply:
To make it simpler I will remove key generation method from this class. If something happens and we will need it, then we will add it ;)

Reply:
SessionId is just id of records from table Session from what I know. And userKey is validation that specific user is logged in.
For me this endpoint is good for now.

Reply:
I already delete it, but I think you are right. It might be unnecessary to generate new session key outside log in. If session expired, you just have to log in again (creating new session and new userKey is generated). I'm just not sure what happen with items in cart in that situation.

Reply:
Ok I think you can merge it, if we will need it in futere then no problem to add it.